src/store/middleware/telemetryMiddleware.js

```diff
@@ -0,0 +1,29 @@
+import{
+    roverPositionReportReceived
+  } from "../teleSlice";
+  import { messageReceivedFromRover } from "../roverSocketSlice";
+
+  const telemetryMiddleware = store => next => action => {
+    const result = next(action);
+    if (action.type === messageReceivedFromRover.type) {
+      if (action.payload.message.type === "poseReport") {
+        const { orientW, orientX, orientY, orientZ, posX, posY, posZ, timestamp } = action.payload.message;
+        store.dispatch(roverPositionReportReceived({
+          poseReport: {
+            orientW,
+            orientX,
+            orientY,
+            orientZ,
+            posX,
+            posY,
+            posZ,
+            timestamp
```

Comment:
Can we name this something else? Like we discussed, this isn't going to be timestamp, it'll be something more like data age or recency.